dist/s_test_suite_no_executable

```diff
@@ -1,9 +1,12 @@
 #!/usr/bin/env bash
 
-cd ../test/suite
-if [[ $(find . -executable -type f | grep -v -w "./run.py") ]]; then
-    echo "The following files should not be executable: " &&
-    find . -executable -type f | grep -v -w "./run.py"
+# All test scripts should be routed through "run.py", running them directly is an anti-pattern.
+# Use the executable bits on the scripts as weak evidence of the anti-pattern.
+workspace=$(git rev-parse --show-toplevel)
+exes=$(find $workspace/test/suite -type f \( -perm -u+x -o -perm -g+x -o -perm -o+x \) -a ! -name 'run.py' -print)
```

Comment:
nit: I'd rather use `-perm +111` instead of `(...)` expression, it works on all supported platforms. But this will do as well.

Reply:
Actually, because `-perm +#` **is** available on our supported platforms, I'd prefer it to be used.

Reply:
From Linux:
```
# +exes=$(find "$workspace"/test/suite -type f -perm +111 ! -name 'run.py' -print)
ip-10-0-101-50:dist [0]$ ./s_test_suite_no_executable
find: invalid mode '+111'
```